eip/EIPS/eip-4337.md

```diff
@@ -96,28 +97,25 @@ interface IWallet {
 The wallet
 * MUST validate the caller is a trusted EntryPoint
 * The requestId is a hash over the userOp (except signature), entryPoint and chainId
-* MUST validate the signature is a valid signature of the `requestId`
+* If the wallet does not support signature aggregation, it MUST validate the signature is a valid signature of the `requestId` 
```

Comment:
Currently, the way for a wallet to signal if it supports aggregation or not is by having getAggregator() method returning a non-zero address.
So currently, a wallet can change its state from aggregated to non-aggregated mode, but it can't arbitrarily accept both.
One way to achieve this is to change it to getAggregator(UserOperation), or at least getAggregator(signature), so that it can decide based on the signature what aggregation scheme to use.
We do need more feedback as whether this is needed.